src/components/HTMLEngineProvider/HTMLRenderers/AnchorRenderer.tsx

```diff
@@ -83,7 +105,19 @@ function AnchorRenderer({tnode, style, key}: AnchorRendererProps) {
             ]}
             key={key}
             // Only pass the press handler for internal links. For public links or whitelisted internal links fallback to default link handling
-            onPress={internalNewExpensifyPath || internalExpensifyPath ? () => openLink(attrHref, environmentURL, isAttachment) : undefined}
+            onPress={
+                internalNewExpensifyPath || internalExpensifyPath
+                    ? () => openLink(attrHref, environmentURL, isAttachment)
+                    : isTourTask && !hasSeenTour
+                    ? () => {
+                          openExternalLink(attrHref);
+                          setSelfTourViewed(isAnonymousUser());
+                          if (viewTourTaskReport && canModifyViewTourTask && canActionViewTourTask) {
+                              completeTask(viewTourTaskReport);
```

Comment:
Even though I added a code to mark the task as complete, `canModifyViewTourTask` is always `false`, even though through FAB.